src/phoenix/server/api/subscriptions.py

```diff
@@ -104,18 +169,88 @@ async def chat_completion(
                     trace_rowid=trace_rowid,
                     span_id=span_id,
                     parent_id=None,
-                    name="AsyncOpenAI.chat.completion.create",
-                    span_kind="LLM",
+                    name=span_name,
+                    span_kind=LLM,
                     start_time=start_time,
                     end_time=end_time,
-                    attributes={},
-                    events=[],
-                    status_code="OK",
-                    status_message="",
-                    cumulative_error_count=0,
+                    attributes=unflatten(attributes.items()),
+                    events=finished_span.events,
+                    status_code=status.status_code.name,
+                    status_message=status.description or "",
+                    cumulative_error_count=int(not status.is_ok),
                     cumulative_llm_token_count_prompt=0,
                     cumulative_llm_token_count_completion=0,
                     llm_token_count_prompt=0,
                     llm_token_count_completion=0,
                 )
             )
+
+
+def _llm_span_kind() -> Iterator[Tuple[str, Any]]:
+    yield OPENINFERENCE_SPAN_KIND, LLM
+
+
+def _input_value_and_mime_type(input: ChatCompletionInput) -> Iterator[Tuple[str, Any]]:
+    yield INPUT_MIME_TYPE, JSON
+    yield INPUT_VALUE, json.dumps(asdict(input), cls=GraphQLInputJSONEncoder)
+
+
+def _output_value_and_mime_type(output: Any) -> Iterator[Tuple[str, Any]]:
+    yield OUTPUT_MIME_TYPE, JSON
+    yield OUTPUT_VALUE, json.dumps(output, cls=ChatCompletionOutputJSONEncoder)
+
+
+def _llm_input_messages(messages: List[ChatCompletionMessageInput]) -> Iterator[Tuple[str, Any]]:
+    for i, message in enumerate(messages):
+        yield f"{LLM_INPUT_MESSAGES}.{i}.{MESSAGE_ROLE}", message.role.value.lower()
+        yield f"{LLM_INPUT_MESSAGES}.{i}.{MESSAGE_CONTENT}", message.content
+
+
+def _llm_output_messages(content: str, role: str) -> Iterator[Tuple[str, Any]]:
+    yield f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}", role
+    yield f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}", content
+
+
+def _hex(number: int) -> str:
+    """
+    Converts an integer to a hexadecimal string.
+    """
+    return hex(number)[2:]
+
+
+def _datetime(*, epoch_nanoseconds: float) -> datetime:
+    """
+    Converts a Unix epoch timestamp in nanoseconds to a datetime.
+    """
+    epoch_seconds = epoch_nanoseconds / 1e9
+    return datetime.fromtimestamp(epoch_seconds)
+
+
+class GraphQLInputJSONEncoder(JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, Enum):
+            return obj.value
+        return super().default(obj)
+
+
+class ChatCompletionOutputJSONEncoder(JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, BaseModel):
+            return obj.model_dump()
```

Comment:
fyi, we have a `jsonify()` helper function in utilities

Reply:
Seems to be breaking when exported over OTel